code/modules/atmospherics/environmental/LINDA_fire.dm

```diff
@@ -136,6 +136,19 @@
 			AT.fire_act(temperature, volume)
 	return
 
+/obj/effect/hotspot/proc/reaction_volume_calculation(datum/gas_mixture/gasmix)
+	if(!istype(gasmix))
+		return FALSE
+
+	var/list/cached_results = gasmix.reaction_results
+	for (var/reaction in cached_results)
+		if(ispath(reaction, /datum/gas_reaction/plasmafire))
+			. += gasmix.reaction_results[/datum/gas_reaction/plasmafire] * FIRE_GROWTH_RATE
+		else if(ispath(reaction, /datum/gas_reaction/tritfire))
+			. += gasmix.reaction_results[/datum/gas_reaction/tritfire] * FIRE_GROWTH_RATE
+		else if(ispath(reaction, /datum/gas_reaction/h2fire))
+			. += gasmix.reaction_results[/datum/gas_reaction/h2fire] * FIRE_GROWTH_RATE
```

Comment:
1: you're caching the results and then not using the cache :sadkirby:
2: You should profile/use a static typecache for this, it'd be faster. And that way you only need 1 line for the reaction addition 

Hell why does this need to be a proc?

Reply:
ok ill inline it